test/NuGet.Core.Tests/NuGet.Build.Tasks.Console.Test/BuildTasksUtilityTests.cs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using FluentAssertions;
+using NuGet.Configuration;
+using NuGet.Test.Utility;
+using Xunit;
+
+namespace NuGet.Build.Tasks.Console.Test
+{
+    public class BuildTasksUtilityTests
+    {
+        [Fact]
+        public void GetSources_WithRestoreSourcesProperty_ResolvesAgainstProjectDirectory()
+        {
+            using (var testDir = TestDirectory.CreateInTemp())
+            {
+                // Arrange
+
+                var startupDirectory = Path.Combine(testDir, "startup");
+                var projectDirectory = Path.Combine(testDir, "project");
+                var relativePath = "relativeSource";
+
+                // Act
+                var effectiveSources = BuildTasksUtility.GetSources(
+                     startupDirectory: startupDirectory,
+                     projectDirectory: projectDirectory,
+                     sources: new string[] { relativePath },
+                     sourcesOverride: null,
+                     additionalProjectSources: new string[] { },
```

Comment:
[CA1825: Avoid zero-length array allocations](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1825). Use `Array.Empty<string>()` instead.

Reply:
Fixed, but these are tests and when you write many tests, it's easier to have a variable string. 

After all, there's a reason why the analyzers are not raising these warnings in test projects :)